diff --git forkSrcPrefix/common.gypi forkDstPrefix/common.gypi
index db162537869..1723def3d52 100644
--- forkSrcPrefix/common.gypi
+++ forkDstPrefix/common.gypi
@@ -511,7 +511,6 @@
       [ 'OS in "linux freebsd openbsd solaris android aix os400 cloudabi openharmony"', {
         'cflags': [ '-Wall', '-Wextra', '-Wno-unused-parameter', ],
         'cflags_cc': [
-          '-fno-rtti',
           '-fno-exceptions',
           '-fno-strict-aliasing',
           '-std=gnu++20',
diff --git forkSrcPrefix/tools/v8_gypfiles/features.gypi forkDstPrefix/tools/v8_gypfiles/features.gypi
index 81272ed99d7..2d774b23e6f 100644
--- forkSrcPrefix/tools/v8_gypfiles/features.gypi
+++ forkDstPrefix/tools/v8_gypfiles/features.gypi
@@ -331,6 +331,9 @@
     # depending on platform.
     'v8_advanced_bigint_algorithms%': 1,
 
+    # Cfx changes
+    'use_rtti%': 1,
+
     # Enable 256-bit long vector re-vectorization pass in WASM compilation pipeline.
     'v8_enable_wasm_simd256_revec%' : 0
   },
