diff --git forkSrcPrefix/deps/icu-small/source/tools/toolutil/pkg_genc.cpp forkDstPrefix/deps/icu-small/source/tools/toolutil/pkg_genc.cpp
index c33343318b3d2da6482a5548a9e810e3b96f6632..f144978f92f02d5855fdd9ce02a427c76beb2566 100644
--- forkSrcPrefix/deps/icu-small/source/tools/toolutil/pkg_genc.cpp
+++ forkDstPrefix/deps/icu-small/source/tools/toolutil/pkg_genc.cpp
@@ -877,7 +877,7 @@ getArchitecture(
                 std::terminate(); // Unreachable.
             }
 #       else
-            *pCPU = IMAGE_FILE_MACHINE_UNKNOWN;
+            *pCPU = IMAGE_FILE_MACHINE_AMD64; // Default to x64 for non-ClangCL compilers.
 #       endif
 #   endif
 #   if defined(_M_IA64) || defined(_M_AMD64) || defined (_M_ARM64)
