diff --git a/deps/v8/src/heap/marking-barrier-inl.h b/deps/v8/src/heap/marking-barrier-inl.h
index 306e37e5d7e..558c221e2dc 100644
--- a/deps/v8/src/heap/marking-barrier-inl.h
+++ b/deps/v8/src/heap/marking-barrier-inl.h
@@ -88,6 +88,8 @@ void MarkingBarrier::MarkValueShared(Tagged<HeapObject> value) {
 
 void MarkingBarrier::MarkValueLocal(Tagged<HeapObject> value) {
   DCHECK(!HeapLayout::InReadOnlySpace(value));
+  // Temporary safety check before we find a way to fix race condition between GC and microtasks
+  if (V8_UNLIKELY(!is_activated_)) return;
   if (is_minor()) {
     // We do not need to insert into RememberedSet<OLD_TO_NEW> here because the
     // C++ marking barrier already does this for us.
